experimental/traffic-portal/src/app/core/cdns/cdn-table/cdn-table.component.ts

```diff
@@ -0,0 +1,262 @@
+			multiRow: false,
+			name: "Clear Queued Updates"
+		},
+		{
+			disabled: (): true => true,
+			href: (selectedRow): string => `/core/cdns/${selectedRow.id}/dnssec-keys` ,
+			name: "Manage DNSSEC Keys"
+		},
+		{
+			disabled: (): true => true,
+			href: (selectedRow): string => `/core/cdns/${selectedRow.id}/federations` ,
+			name: "Manage Federations"
+		},
+		{
+			disabled: (): true => true,
+			href: (selectedRow): string => `/core/cdns/${selectedRow.id}/delivery-services` ,
+			name: "Manage Delivery Services"
+		},
+		{
+			href: (selectedRow): string => `/core/profiles?cdnName=${selectedRow.name}` ,
+			name: "Manage Profiles"
+		},
+		{
+			disabled: (): true => true,
+			href: (selectedRow): string => `/core/servers?cdnName=${selectedRow.name}` ,
+			name: "Manage Servers"
+		},
+		{
+			disabled: (): true => true,
+			href: (selectedRow): string => `/core/cdns/${selectedRow.id}/notifications` ,
+			name: "Manage Notifications"
+		},
+	];
+
+	/**
+	 * A subject that child components can subscribe to for access to the fuzzy
+	 * search query text.
+	 */
+	public fuzzySubject: BehaviorSubject<string>;
+
+	/** Form controller for the user search input. */
+	public fuzzControl: FormControl = new FormControl("");
+
+	constructor(
+		private readonly alerts: AlertService,
+		private readonly api: CDNService,
+		public readonly auth: CurrentUserService,
+		private readonly dialog: MatDialog,
+		private readonly route: ActivatedRoute,
+	) {
+		this.fuzzySubject = new BehaviorSubject<string>("");
+		this.cdns = this.api.getCDNs();
+	}
+
+	/** Initializes table data, loading it from Traffic Ops. */
+	public ngOnInit(): void {
+		this.route.queryParamMap.subscribe(
+			m => {
+				const search = m.get("search");
+				if (search) {
+					this.fuzzControl.setValue(decodeURIComponent(search));
+					this.updateURL();
+				}
+			}
+		);
+	}
+
+	/** Update the URL's 'search' query parameter for the user's search input. */
+	public updateURL(): void {
+		this.fuzzySubject.next(this.fuzzControl.value);
+	}
+
+	/**
+	 * Queues or clears updates on a group of CDNs.
+	 *
+	 * @param cdn The CDN on which to queue updates.
+	 * @param queue Whether updates should be queued (`true`) or cleared
+	 * (`false`).
+	 */
+	private async queueUpdates(cdn: ResponseCDN | number, queue: boolean = true): Promise<void> {
+		if (typeof cdn === "number") {
+			cdn = await this.api.getCDNs(cdn);
+		}
+		const title = `${queue ? "Queue" : "Clear"} Updates on ${cdn.name}?`;
+		const action = queue ? "queue" : "dequeue";
+		const ref = this.dialog.open<DecisionDialogComponent, DecisionDialogData, boolean>(DecisionDialogComponent, {
+			data: {
+				message: `Are you sure you want to ${action} server updates for all of the ${cdn.name} servers?`,
+				title,
+			}
+		});
+		if (!await ref.afterClosed().toPromise()) {
+			return;
+		}
+		await this.api.queueCDNUpdates(cdn, action);
+		this.alerts.newAlert(
+			AlertLevel.SUCCESS,
+			"Queued CDN server updates",
```

Comment:
this isn't always true. When I clear updates on a CDN, it tells me "Queued CDN server updates", which makes me think I accidentally clicked the wrong button

Reply:
No longer hard-coding the action in 3dd580cbb0